packages/services/src/simulated-trade-execution.service.ts

```diff
@@ -170,8 +165,11 @@ export class SimulatedTradeExecutionService {
         chainOptions,
       );
 
-      // Validate cross-chain trading rules
-      this.validateCrossChainTrading(chainInfo);
+      // Validate cross-chain trading rules using competition-specific settings
+      this.validateCrossChainTrading(
+        chainInfo,
+        competition.crossChainTradingType,
```

Comment:
since the comp stores the feature, we can just use the db value instead of relying on the constructor value. 

note: "features" is probably a bit of a misnomer since it's not really a feature flag, but an explicit and overridable runtime value. it should probably be a zod default in the admin "create competition" endpoint, and that's it.

created a follow up: https://linear.app/recall-labs/issue/APP-588/remove-competition-app-feature-flags